atom/browser/web_contents_preferences.cc

```diff
@@ -160,7 +160,14 @@ void WebContentsPreferences::AppendExtraCommandLineSwitches(
   if (window) {
     bool visible = window->IsVisible() && !window->IsMinimized();
     if (!visible)  // Default state is visible.
-      command_line->AppendSwitch("hidden-page");
+      command_line->AppendSwitch(switches::kHiddenPage);
+  } else {
+    // Inherit initial visibilty state from parent window in webviews
+    bool hidden_page;
+    if (web_preferences.GetBoolean(options::kHiddenPage, &hidden_page) &&
```

Comment:
I tend to implement this with native code instead of adding a new param to guest contents, every time we do so it would make code harder to maintain.

The relationship between host and webview WebContents is maintained by `WebViewManager`, we can add a new API there to receive the host WebContents of a webview, and then reuse the logic for normal WebContents.